dali/operators/reader/loader/numpy_loader.cc

```diff
@@ -38,6 +37,75 @@ TypeInfo TypeFromNumpyStr(const std::string &format) {
   DALI_FAIL("Unknown Numpy type string");
 }
 
+void ParseHeaderMetadata(NumpyParseTarget& target, std::string header) {
+  header.erase(std::remove_if(header.begin(), header.end(), ::isspace), header.end());
+
+  const char t1[] = "{'descr':\'";
+  size_t l1 = strlen(t1);
+  DALI_ENFORCE(strncmp(header.c_str(), t1, l1) == 0);
+  size_t pos = l1;
+  auto iter = header.find('\'', pos);
+  DALI_ENFORCE(iter != std::string::npos);
+
+  std::string typestring = header.substr(pos, iter - pos);
+  pos = iter + 1;
+
+  // < means LE, | means N/A, = means native. In all those cases, we can read
+  bool little_endian =
+    (typestring[0] == '<' || typestring[0] == '|' || typestring[0] == '=');
```

Comment:
Maybe I'm nitpicky, but do you think we can check this under the `l1` and create the `tid` string once? Instead of creating the string with this character in front and creating next string with 1 character less?